static/script.js

```diff
@@ -41,19 +41,22 @@ function pageSetup() {
     });
 
     document.getElementById('submit-button').addEventListener('click', function(event) {
-        let filesContent = getFilesContent();
-
-        if (filesContent.length <= 0) {
-            showUserInfo("Please add a file first");
+        if(!alertIfFileContainsPassword()){
             return;
         }
 
-        let formData = new FormData();
-        filesContent.forEach(element => {
-            formData.append('files[]', element);
-        });
+            let filesContent = getFilesContent();
+            if (filesContent.length <= 0) {
+                showUserInfo("Please add a file first");
+                return;
+            }
 
-        fetch('/multipart', {
+            let formData = new FormData();
+            filesContent.forEach(element => {
+                formData.append('files[]', element);
+            });
+            
+            fetch('/multipart', {
```

Comment:
Just one small last nitpick, don't change the indentation now, makes the diff also a lot cleaner then